test/e2e/artifacts_test.go

```diff
@@ -352,7 +368,7 @@ func (s *ArtifactsSuite) TestArtifactGC() {
 				}
 			})
 
-		if when.WorkflowCondition(func(wf *wfv1.Workflow) bool {
+		if tt.workflowShouldSucceed && when.WorkflowCondition(func(wf *wfv1.Workflow) bool {
```

Comment:
was previously trying to avoid intermittent test failures due to the workflow itself failing when it shouldn't - but now there's a Workflow which fails in here so this is how I'm allowing that one to be failed